### PR TITLE
Reports for contentMetadata imageData element incomplete or missing or on wrong type

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

part of #3446

we need data to determine what we want to do with missing or partially missing imageData, and when it's put on resources of the wrong type (e.g. the resource type is wrong in the contentMetadata)

## How was this change tested?

rails console.  Running reports.

### CODE REVIEW

**Review by justinlittman — COMMENTED** [review]
Did you decide that page types needed to be taken into account?

**Review by jcoyne — APPROVED** [review]

File: bin/reports/report-content-missing-image-data [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# frozen_string_literal: true
+
+require_relative '../../config/environment'
+require_relative '../../lib/report'
+
+Report.new(name: 'content-missing-image-data', dsid: 'contentMetadata').run do |ng_xml|
+  # imageData child element missing from file child of a resource of type image
+  ng_xml.xpath('/contentMetadata/resource[@type="image"]/file[not(./imageData)]').blank? ||
+
+    # imageData width attribute is missing
```
[the remark this answers is not in the document]
Comment (ndushay) [reply bin/reports/report-content-missing-image-data]: Thanks!   That's much nicer (and I should have figured it out, but didn't).  Refactored to make it even more readable.

**Comment by ndushay** [comment]
Thanks for the approval;  I'll merge when I'm sure the reports are getting the data we want.

**Comment by ndushay** [comment]
### Report on image/jp2 files with imageData missing height or width info

There are none.  Or my report is wrong.

**Comment by ndushay** [comment]
I'm declaring victory here and in #3446 and I'm merging this and closing that ticket.

**Pull request merged**